### PR TITLE
Add DGCNN demo to Read the Docs

### PR DESCRIPTION
**Pull request by huonw**
This adds the notebook to be rendered on Read the Docs and adjusts the graph classification index to match.

Graph classification index: https://stellargraph--1450.org.readthedocs.build/en/1450/demos/graph-classification/index.html
Notebook: https://stellargraph--1450.org.readthedocs.build/en/1450/demos/graph-classification/dgcnn-graph-classification.html

See: #1449

### CODE REVIEW

File: docs/demos/graph-classification/index.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Graph classification using StellarGraph
 =======================================
 
-`StellarGraph <https://github.com/stellargraph/stellargraph>`_ provides an algorithm for graph classification. This folder contains a demo to explain how it works and how to use it as part of a TensorFlow Keras data science workflow.
+`StellarGraph <https://github.com/stellargraph/stellargraph>`_ provides algorithms for graph classification. This folder contains a demo to explain how they works and how to use them as part of a TensorFlow Keras data science workflow.
 
 A graph classification task predicts an attribute of each graph in a collection of graphs. For instance, labelling each graph with a categorical class (binary classification or multiclass classification), or predicting a continuous number (regression). It is supervised or semi-supervised, where the model is trained using a subset of graphs that have ground-truth labels.
```
Comment (PantelisElinas) [inline docs/demos/graph-classification/index.txt:6]: Both algorithms are supervised and neither is semi-supervised. The latter means that some or all of the unlabelled graphs are used in training together with the labelled ones. However, the unlabelled graphs are not used during training for either of the two graph classification algorithms.

File: docs/demos/graph-classification/index.txt [hunk 2]
```diff
@@ -1,7 +1,7 @@
 Graph classification using StellarGraph
 =======================================
 
-`StellarGraph <https://github.com/stellargraph/stellargraph>`_ provides an algorithm for graph classification. This folder contains a demo to explain how it works and how to use it as part of a TensorFlow Keras data science workflow.
+`StellarGraph <https://github.com/stellargraph/stellargraph>`_ provides algorithms for graph classification. This folder contains a demo to explain how they works and how to use them as part of a TensorFlow Keras data science workflow.
```
Comment (PantelisElinas) [inline docs/demos/graph-classification/index.txt:4]: "they works" to "they work".

Comment (huonw) [reply docs/demos/graph-classification/index.txt:4]:
```suggestion
`StellarGraph <https://github.com/stellargraph/stellargraph>`_ provides algorithms for graph classification. This folder contains a demo to explain how they work and how to use them as part of a TensorFlow Keras data science workflow.
```

File: docs/demos/graph-classification/index.txt (hunk 1 above)
Comment (huonw) [reply docs/demos/graph-classification/index.txt:6]:
```suggestion
A graph classification task predicts an attribute of each graph in a collection of graphs. For instance, labelling each graph with a categorical class (binary classification or multiclass classification), or predicting a continuous number (regression). It is supervised, where the model is trained using a subset of graphs that have ground-truth labels.
```

**Pull request merged**